### PR TITLE
Implementing support for decimal type

### PR DESCRIPTION
**Pull request by WWW0030**

### CODE REVIEW

**Review by mewim — CHANGES_REQUESTED** [review]
You should use a proper decimal pacakge https://pkg.go.dev/github.com/shopspring/decimal

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,10 @@ go 1.20
 
 require github.com/google/uuid v1.6.0
 
-require github.com/stretchr/testify v1.9.0
+require (
```
Comment (mewim) [inline go.mod:7]: I would prefer seperate the require lines to oone dependency per line

File: value_helper.go [hunk 2]
```diff
@@ -424,6 +425,15 @@ func kuzuValueToGoValue(kuzuValue C.kuzu_value) (any, error) {
 		return kuzuStructValueToGoValue(kuzuValue)
 	case C.KUZU_MAP:
 		return kuzuMapValueToGoValue(kuzuValue)
+	case C.KUZU_DECIMAL:
+		var outString *C.char
+		status := C.kuzu_value_get_decimal_as_string(&kuzuValue, &outString)
+		if status != C.KuzuSuccess {
+			return nil, fmt.Errorf("failed to get string value of decimal type with status: %d", status)
+		}
+		goDecimal, _ := decimal.NewFromString(C.GoString(outString))
```
Comment (mewim) [inline value_helper.go:434]: You should not ignore the error in this case. If it fails, you should pass the error to the user, something like `failed to convert convert decimal value with error %s`.

File: value_helper.go [hunk 3]
```diff
@@ -424,6 +425,15 @@ func kuzuValueToGoValue(kuzuValue C.kuzu_value) (any, error) {
...
+		defer C.kuzu_destroy_string(outString)
```
Comment (mewim) [inline value_helper.go:435]: This does not require a deferred destroy. It can be destroyed immediately at this point.

File: value_test.go [hunk 4]
```diff
@@ -348,6 +349,27 @@ func TestMap(t *testing.T) {
...
+
+	expected := decimal.NewFromFloat(5.7)
+	assert.Equal(t, expected, value.([]interface{})[0])
+	expected = decimal.NewFromFloat(8.3)
+	assert.Equal(t, expected, value.([]interface{})[1])
+	expected = decimal.NewFromFloat(8.7)
+	assert.Equal(t, expected, value.([]interface{})[2])
+	expected = decimal.NewFromFloat(13.7)
+	assert.Equal(t, expected, value.([]interface{})[3])
+	expected = decimal.NewFromFloat(8.7 * 13.7)
```
Comment (mewim) [inline value_test.go:370]: There are multiple calls to `value.([]interface{})`. This is not readable and redundent. You should assign it to a var and use it for the following assertions.

**Review by mewim — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (WWW0030) [reply go.mod:7]: Oops, looks like there is more than one file with the require problem, let me push a new change

File: value_test.go [hunk 5]
```diff
@@ -348,6 +349,23 @@ func TestMap(t *testing.T) {
...
 }
 
+func TestDecimal(t *testing.T) {
+	_, conn := SetupTestDatabase(t)
+	res, error := conn.Query("UNWIND [1] AS A UNWIND [5.7, 8.3, 8.7, 13.7] AS B WITH cast(CAST(A AS DECIMAL) * CAST(B AS DECIMAL) AS DECIMAL(18, 1)) AS PROD RETURN COLLECT(PROD) AS RES")
+	assert.Nil(t, error)
+	assert.True(t, res.HasNext())
+	next, _ := res.Next()
+	value, _ := next.GetValue(0)
+	interfaceValues, _ := value.([]interface{})
```
Comment (mewim) [inline value_test.go:359]: `interfaceValues` is not a good name. Maybe call it `valueList` or `valueSlice`

**Review by mewim — APPROVED** [review]

**Pull request merged**